src/prefect/blocks/notifications.py

```diff
@@ -400,3 +401,77 @@ def block_initialization(self) -> None:
             ).url()
         )
         self._start_apprise_client(url)
+
+
+class MattermostWebhook(AbstractAppriseNotificationBlock):
+    """
+    Enables sending notifications via a provided Mattermost webhook.
+    See [Apprise notify_Mattermost docs](https://github.com/caronc/apprise/wiki/Notify_Mattermost) # noqa
+
+
+    Examples:
+        Load a saved Mattermost webhook and send a message:
+        ```python
+        from prefect.blocks.notifications import MattermostWebhook
+
+        Mattermost_webhook_block = MattermostWebhook.load("BLOCK_NAME")
+
+        Mattermost_webhook_block.notify("Hello from Prefect!")
+        ```
+    """
+
+    _description = "Enables sending notifications via a provided Mattermost webhook."
+    _block_type_name = "Mattermost Webhook"
+    _block_type_slug = "Mattermost-webhook"
+    _logo_url = "https://images.ctfassets.net/zscdif0zqppk/3mlbsJDAmK402ER1sf0zUF/a48ac43fa38f395dd5f56c6ed29f22bb/mattermost-logo-png-transparent.png?h=250"
+    _documentation_url = "https://docs.prefect.io/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.MattermostWebhook"
+
+    hostname: str = Field(
+        default=...,
+        description="The hostname of your Mattermost server.",
+        example="Mattermost.example.com",
+    )
+
+    token: SecretStr = Field(
+        default=...,
+        description="The token associated with your Mattermost webhook.",
+    )
+
+    botname: Optional[str] = Field(
+        default=None,
+        description="The name of the bot that will send the message.",
+    )
+
+    channels: Optional[List[str]] = Field(
+        default=None,
+        description="The channel(s) you wish to notify.",
+    )
+
+    include_image: bool = Field(
+        default=False,
+        description="Whether to include the Apprise status image in the message.",
+    )
+
+    fullpath: Optional[str] = Field(
```

Comment:
`hostname` is required, but `fullpath` is not. Would it be redundant to provide both the hostname and a full path to the server? Would it be more clear there was only one way to configure this?

Reply:
`hostname` and `fullpath` are [different things that are both used in constructing the url](https://github.com/caronc/apprise/blob/master/apprise/plugins/NotifyMattermost.py#L313) so i dont think itd be redundant to pass both

Reply:
Oh I see, I agree that path and the description that you added make the purpose of this field more clear.